src/sentry/api/endpoints/internal/rpc.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from django.conf import settings
+from OpenSSL.crypto import X509Name
+from rest_framework import permissions
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.authentication import MutualTlsAuthentication
+from sentry.api.base import Endpoint, control_silo_endpoint, region_silo_endpoint
+from sentry.services.hybrid_cloud.rpc import ServiceEndpoint, endpoint_registry
+from sentry.services.hybrid_cloud.rpc.endpoints import MethodEndpoint
+from sentry.silo import SiloLimit, SiloMode
+
+
+class SiloPermission(permissions.BasePermission):
+    def has_permission(self, request: Request, view: RpcEndpoint):
+        auth: X509Name = request.auth
+        if not isinstance(request.auth, X509Name):
```

Comment:
Presumably this happens on the LB layer, right?

Reply:
Yes, from my understanding of the k8s docs, we can have a specific k8s ingress-controller that validates certs for the path used by this RPC endpoint.  It would be good to check the CN/DN of the cert in application code as well though.